src/som/interpreter/actors/EventualMessage.java

```diff
@@ -79,16 +80,16 @@ public SourceSection getTargetSourceSection() {
     return onReceive.getRootNode().getSourceSection();
   }
 
-  public long serialize(final SnapshotBuffer sb) {
-    ReceivedRootNode rm = (ReceivedRootNode) this.onReceive.getRootNode();
-
-    if (sb.needsToBeSnapshot(getMessageId())) {
-      // Not sure if this is optimized, worst case need to duplicate this for all messages
-      return rm.getSerializer().execute(this, sb);
-    } else {
-      // need to be careful, might interfere with promise serialization...
-      return -1;
+  @TruffleBoundary
+  // TODO: can we establish a structure for this? at the moment, we have an
```

Comment:
This is now handled, right?

Reply:
The serializer is compilation final (child node of receive).
There is still some indirection of accessing fields